composer/datasets/streaming/world.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2022 MosaicML. All Rights Reserved.
+
+"""The :class:`World` class is used for easily querying distributed training info, used by `StreamingDataset`.
+"""
+
+from typing import NamedTuple
+
+from torch.utils.data import get_worker_info
+
+from composer.utils import dist
+
+__all__ = ["World", "get_world"]
+
+
+class World(NamedTuple):
```

Comment:
For readability, can you use the same termonology as `composer.utils.dist` where the names do align?

`global_node` -> `node_rank`
`global_num_nodes` -> `node_world_size`
`global_device` -> `global_rank` (or `rank`)
`global_num_devices` -> `world_size`
`node_device` -> `local_rank`
`node_num_devices` -> `local_world_size`

For the naming of workers, maybe match `torch.utils.data.get_worker_info` (and use `rank` instead of `device`?) Agree that "local" doesn't make sense, as that is ambiguous for node vs rank.

`global_worker` -> `global_worker_id`
`node_worker` -> `node_worker_id`
`device_worker` -> `rank_worker_id`
`device_num_workers` -> `rank_num_workers`

Reply:
I'm gonna push back pretty hard on this... I agree that consistency is good, but if anything the left side is what we should use everywhere, not the right. The `world_size` and `rank` terminology is a lot less clear than the `global/node/device/worker` hierarchy.

I'll make a JIRA issue to reolve `World` and `composer.utils.dist1, but I would like to keep it as is here.

Reply:
PyTorch uses this terminology -- see https://pytorch.org/docs/stable/elastic/run.html#launcher-api
I think we should follow that for anything node/rank related, and then follow the `torch.utils.data.get_worker_info` for the naming of the worker variables.

Reply:
Not sure if we should follow PyTorch convention for its own sake, if you take a look at `get_partition` the readability is greatly _improved_ with the new hierarchy, and I think we ought to go for readability > consistency.

I think this is a situation similar to batch/minibatch/microbatch, where the current default naming that PyTorch uses is legacy from a time when people were not doing comlex distributed training.

E.g, how many ML users are aware that the DataLoader `batch_size` is really the `device_batch_size`? Is the legacy naming more or less clear than what it could be?